The tests use at least 2 GPUs but don't check for it failing with
> torch.AcceleratorError: CUDA error: invalid device ordinal

See https://github.com/pytorch/pytorch/pull/189404

Author: Alexander Grund (TU Dresden)
diff --git a/test/distributed/optim/test_zero_redundancy_optimizer.py b/test/distributed/optim/test_zero_redundancy_optimizer.py
--- a/test/distributed/optim/test_zero_redundancy_optimizer.py
+++ b/test/distributed/optim/test_zero_redundancy_optimizer.py
@@ -470,6 +470,7 @@ class TestZeroRedundancyOptimizerDistributed(TestZeroRedundancyOptimizer):
             s2.step()
             self.assertEqual(x, x2)
 
+    @skip_if_no_gpu
     def test_sharding(self):
         """
         Check ZeroRedundancyOptimizer's parameter sharding at construction
@@ -492,6 +493,7 @@ class TestZeroRedundancyOptimizerDistributed(TestZeroRedundancyOptimizer):
             sum(sizes),
         )
 
+    @skip_if_no_gpu
     def test_add_param_group(self):
         """Check that ZeroRedundancyOptimizer properly handles adding a new
         parameter group a posteriori and that all ranks get a shard of the
